### PR TITLE
Feature: show only connector

### PR DESCRIPTION
**Pull request by sivaRajanaDivami**
#TITLE: Filtered List of Connector Types in the Home page to display only the ones which have connections

## Description
Summary of changes

## PR Type
- [ ] Bug Fix
- [x] Feature
- [ ] Build Fix
showonly

- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [Jira issue #](fill-in.org)

### CODE REVIEW

File: app/cdap/components/ConnectionList/Components/SubHeader/index.tsx [hunk 1]
```diff
@@ -33,18 +34,18 @@ export default function SubHeader() {
           <Link color="inherit" to={`/ns/${getCurrentNamespace()}/home`}>
             Home
           </Link>
-          <Typography>Data Sources</Typography>
+          <Typography className={classes.breadCrumbTyporgraphy}>Data Sources</Typography>
```
Comment (adityadivami) [inline app/cdap/components/ConnectionList/Components/SubHeader/index.tsx]: rename to classes.breadcrumbTypography

File: app/cdap/components/ConnectionList/Components/SubHeader/index.tsx [hunk 2]
```diff
@@ -23,6 +23,7 @@ import { getCurrentNamespace } from 'services/NamespaceStore';
 import NavigateNextIcon from '@material-ui/icons/NavigateNext';
 import AddCircleOutlineOutlinedIcon from '@material-ui/icons/AddCircleOutlineOutlined';
 import SaveAltRoundedIcon from '@material-ui/icons/SaveAltRounded';
+import { ADD_CONNECTION_LABEL, IMPORT_DATA_LABEL } from './constants';
 
 export default function SubHeader() {
```
Comment (adityadivami) [inline app/cdap/components/ConnectionList/Components/SubHeader/index.tsx]: remove function name

File: app/cdap/components/ConnectionList/Components/SubHeader/styles.ts [hunk 3]
```diff
@@ -63,7 +63,7 @@ export const useStyles = makeStyles({
     fontSize: '14px',
   },
   breadCrumbTyporgraphy: {
-    color: '#000000',
+    color: '#212121',
```
Comment (adityadivami) [inline app/cdap/components/ConnectionList/Components/SubHeader/styles.ts]: use Mui grey[900]

File: app/cdap/components/WrangleHome/Components/WrangleHomeTitle/styles.tsx [hunk 4]
```diff
@@ -22,7 +22,7 @@ export const useStyles = makeStyles({
     lineHeight: '30px',
     letterSpacing: '0.15px',
     marginBottom: '-4px',
-    color: '#000000',
+    color: '#212121',
```
Comment (adityadivami) [inline app/cdap/components/WrangleHome/Components/WrangleHomeTitle/styles.tsx]: use Mui grey[900]

File: app/cdap/components/WrangleHome/index.tsx [hunk 5]
```diff
@@ -23,6 +23,7 @@ import WrangleCard from './Components/WrangleCard';
 import WrangleHomeTitle from './Components/WrangleHomeTitle';
 import { GradientLine, HeaderImage } from './icons';
 import { useStyles } from './styles';
+import { CONNECTORS_TITLE, DATA_EXPLORATION_TITLE, VIEW_ALL_LABEL } from './constants';
 
 export default function WranglerHome() {
```
Comment (adityadivami) [inline app/cdap/components/WrangleHome/index.tsx]: remove function name

write export default function() {

//component code
}

**Review by adityadivami — CHANGES_REQUESTED** [review]

File: app/cdap/components/ConnectionList/Components/SubHeader/index.tsx (hunk 2 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/ConnectionList/Components/SubHeader/index.tsx]: done

File: app/cdap/components/ConnectionList/Components/SubHeader/index.tsx (hunk 1 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/ConnectionList/Components/SubHeader/index.tsx]: done

File: app/cdap/components/ConnectionList/Components/SubHeader/styles.ts (hunk 3 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/ConnectionList/Components/SubHeader/styles.ts]: done

File: app/cdap/components/WrangleHome/Components/WrangleHomeTitle/styles.tsx (hunk 4 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/WrangleHome/Components/WrangleHomeTitle/styles.tsx]: done

File: app/cdap/components/WrangleHome/index.tsx (hunk 5 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/WrangleHome/index.tsx]: done

**Review by adityadivami — APPROVED** [review]

File: app/cdap/components/ConnectionList/Components/SubHeader/styles.ts [hunk 6]
```diff
@@ -62,8 +62,8 @@ export const useStyles = makeStyles({
     alignItems: 'flex-end',
     fontSize: '14px',
   },
-  breadCrumbTyporgraphy: {
-    color: '#000000',
+  breadcrumbTyporgraphy: {
```
Comment (njbriggs) [inline app/cdap/components/ConnectionList/Components/SubHeader/styles.ts:65]: Typo: `Typography`

File: app/cdap/components/ConnectionList/Components/SubHeader/constants.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+export const ADD_CONNECTION_LABEL = 'Add connection';
+export const IMPORT_DATA_LABEL = 'Import data';
```
Comment (njbriggs) [inline app/cdap/components/ConnectionList/Components/SubHeader/constants.ts:18]: Please `I18n` these in `text-en.yaml`

File: app/cdap/components/WrangleHome/constants.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+export const VIEW_ALL_LABEL = 'View all';
+export const DATA_EXPLORATION_TITLE =
+  'Continue ongoing data explorations, pick up where you left off';
+export const CONNECTORS_TITLE = 'Start data exploration';
```
Comment (njbriggs) [inline app/cdap/components/WrangleHome/constants.ts:20]: Please I18n these

File: app/cdap/components/WrangleHome/Components/WrangleCard/index.tsx [hunk 9]
```diff
@@ -43,6 +44,11 @@ export default function WrangleCard() {
   // then using unshift function to add an object for Imported Dataset to entire ConnectorTypes Array.
   const getConnectorTypesNames = async () => {
     let fetchedConnectorTypes = await fetchConnectors();
+    const categorizedConnections = await getCategorizedConnections();
+    const connectorTypeWithConnections = [];
+    categorizedConnections.forEach((itemEach, key) => {
+      connectorTypeWithConnections.push(key);
```
Comment (njbriggs) [inline app/cdap/components/WrangleHome/Components/WrangleCard/index.tsx:50]: Why not make `connectorTypeWithConnections` an Object or Map and save on the `find` on line 130?

File: app/cdap/components/ConnectionList/Components/SubHeader/styles.ts (hunk 6 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/ConnectionList/Components/SubHeader/styles.ts:65]: done

File: app/cdap/components/ConnectionList/Components/SubHeader/constants.ts (hunk 7 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/ConnectionList/Components/SubHeader/constants.ts:18]: Done

File: app/cdap/components/WrangleHome/constants.ts (hunk 8 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/WrangleHome/constants.ts:20]: Done

File: app/cdap/components/WrangleHome/Components/WrangleCard/index.tsx (hunk 9 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/WrangleHome/Components/WrangleCard/index.tsx:50]: Done

**Pull request closed without merging**